test/addons/rook-external-cluster/start

```diff
@@ -0,0 +1,176 @@
+
+    if run_script_result.returncode != 0:
+        print("Failed to run script: " + run_script_result.stderr)
+        sys.exit(1)
+
+    print("Script output: " + run_script_result.stdout)
+    sys.exit(0)
+
+    os.unlink(script_name)
+
+def get_ceph_toolbox_pod_name(cluster):
+    return kubectl.get(
+        "pod",
+        "-l app=rook-ceph-tools",
+        "--output=jsonpath={.items[0].metadata.name}",
+        f"--namespace={ROOK_CEPH_NAME_SPACE}",
+        context=cluster
+    )
+
+def get_rook_ceph_external_cluster_details(podname, cluster):
+    try:
+        out = kubectl.exec(
+            podname,
+            f"--namespace={ROOK_CEPH_NAME_SPACE}",
+            "--",
+            "curl",
+            "-o",
+            "/tmp/create-external-cluster-resources.py",
+            "https://raw.githubusercontent.com/rook/rook/master/deploy/examples/create-external-cluster-resources.py",
+            context=cluster
+        )
+    except commands.Error as e:
+        logger.error(f"kubectl.exec failed: {e}")
+        raise
+
+    try:
+        out = kubectl.exec(
+            podname,
+            f"--namespace={ROOK_CEPH_NAME_SPACE}",
+            "--",
+            "python3",
+            "/tmp/create-external-cluster-resources.py",
+            "--format",
+            "bash",
+            "--rbd-data-pool-name",
+            "replicapool",
+            # "--cephfs-filesystem-name",
+            # "cephfs",
+            context=cluster
+        )
+    except commands.Error as e:
+        logger.error(f"kubectl.exec failed: {e}")
+        raise
+
+    return out
+
+def deploy(cluster):
+    print("Deploying rook ceph cluster in external mode")
+    with drenv.kustomization(
+        "kustomization.yaml",
+        base_url=BASE_URL,
+    ) as kustomization:
+        kubectl.apply("--kustomize", kustomization, context=cluster)
+
+def wait(cluster):
+    print("Waiting until rook ceph external cluster is ready")
+    drenv.wait_for(
+        "cephcluster/my-cluster",
+        output="jsonpath={.status.phase}",
+        namespace="rook-ceph",
+        profile=cluster,
+    )
+    kubectl.wait(
+        "cephcluster/my-cluster",
+        "--for=jsonpath={.status.phase}=Ready",
+        "--namespace=rook-ceph",
+        "--timeout=300s",
+        context=cluster,
+    )
+
+
+if len(sys.argv) != 3:
+    print(f"Usage: {sys.argv[0]} cluster external_storage_cluster")
+    sys.exit(1)
+
+os.chdir(os.path.dirname(__file__))
+cluster = sys.argv[1]
+external_storage_cluster = sys.argv[2]
+
+wait(external_storage_cluster)
+deploy(cluster)
+podname = get_ceph_toolbox_pod_name(external_storage_cluster)
+print(f"podname {podname}")
+cluster_details = get_rook_ceph_external_cluster_details(podname, external_storage_cluster)
+print(f"cluster details {cluster_details}")
+rook_connect_to_external_storage_cluster(cluster_details, cluster)
+wait(cluster)
```

Comment:
Almost all script use `deploy()` to deploy stuff, and `wait()` for waiting until deployed stuff is ready. Here we use them in another way, and add extra steps after that. It will be better call:

    deploy(cluster, external_cluster)
    wait(cluster, external_clsuter)

And put keep the details inside the standard functions.

It looks like the actual code can be something like:

    def deploy():
        wait for external cluster
        deploy customization

    def wait():
        wait for local cluster